app/classifier/tasks/shortcut-editor.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+import Shortcut from './shortcut';
+
+export default class ShortcutEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.addAnswer = this.addAnswer.bind(this);
+    this.toggleShortcut = this.toggleShortcut.bind(this);
+  }
+
+  toggleShortcut(e) {
+    let nextTaskID;
+
+    if (e.target.checked) {
+      const taskCount = Object.keys(this.props.workflow.tasks).length;
+
+      let taskIDNumber = -1;
+
+      while (!((nextTaskID != null) && !(nextTaskID in this.props.workflow.tasks))) {
+        taskIDNumber += 1;
+        nextTaskID = `T${taskCount + taskIDNumber}`;
```

Comment:
So, if `addNewTask` is bound to the parent and passed down as a prop, then the task prop in the shortcut-editor will change when the function is called and line 29 in shortcut-editor (`this.props.task.unlinkedTask = nextTaskID;`) will be set on the shortcut task rather than the original task it needs to be linked to. I tried passing the function itself as a prop (`addNewTask={this.addNewTask}`), but the functionality on the child component was being wonky.